src/python/pants/engine/objects.py

```diff
@@ -175,6 +175,9 @@ def __iter__(self) -> Iterator[_C]:
     def __bool__(self) -> bool:
         return bool(self.dependencies)
 
+    def __repr__(self) -> str:
+        return repr(self.dependencies)
```

Comment:
See https://github.com/pantsbuild/pants/pull/9437 for an alternative implementation. Do you mind removing this please to reduce merge conflicts?

Reply:
Sure, this was just for debugging.